static/js/stats/stats.js

```diff
@@ -282,12 +290,42 @@ function populate_messages_sent_over_time(data) {
     }
 }
 
+function format_date_for_last_update(last_update) {
```

Comment:
ah, sorry for being unclear :|. More important is to get the locale behavior correct than to get the exact formatting correct. E.g. https://www.w3schools.com/jsref/jsref_tolocalestring.asp is one way to get a locale-sensitive string. (Not sure if there are other more flexible ways, but that one is good enough if there isn't.)